### PR TITLE
CEDS 1283 - Holder of Authorisation AutoComplete

### PR DESCRIPTION
**Pull request by ed0906**
As part of this I've moved all code-lists into a code list directory as the conf directory was getting a little crowded

### CODE REVIEW

File: test/config/AppConfigSpec.scala [hunk 1]
```diff
@@ -150,11 +150,11 @@ class AppConfigSpec extends CustomExportsBaseSpec {
     }
 
     "have countryCodesJsonFilename" in {
-      validConfigService.countryCodesJsonFilename must be("location-autocomplete-canonical-list.json")
+      validConfigService.countryCodesJsonFilename must be("code-lists/location-autocomplete-canonical-list.json")
```
Comment (benjaminparker) [inline test/config/AppConfigSpec.scala]: why are we having json source and a csv source for countries -

File: app/services/HolderOfAuthenticationCode.scala [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import utils.FileUtil
+
+case class HolderOfAuthenticationCode(value: String)
```
Comment (benjaminparker) [inline app/services/HolderOfAuthenticationCode.scala]: Is it a holder of AUTHENTICATION or holder of AUTHORISATION - a bit confused?

File: test/services/HolderOfAuthenticationCodeSpec.scala [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+package services
+
+import uk.gov.hmrc.play.test.UnitSpec
+
+class HolderOfAuthenticationCodeSpec extends UnitSpec {
+
+  "Holder of Authentication Code" should {
+    "read from file" in {
+      val codes = HolderOfAuthenticationCode.all
+      codes should contain(HolderOfAuthenticationCode("ACE")) // First in file
```
Comment (benjaminparker) [inline test/services/HolderOfAuthenticationCodeSpec.scala]: Remind me to show you the scala test asserts for contains inOrder.... it's pretty cool and a bit stronger than this

File: test/config/AppConfigSpec.scala (hunk 1 above)
Comment (ed0906) [reply test/config/AppConfigSpec.scala]: Beyond the scope of this story, we can raise a tech debt ticket if you like

File: app/services/HolderOfAuthenticationCode.scala (hunk 2 above)
Comment (ed0906) [reply app/services/HolderOfAuthenticationCode.scala]: Fixed

File: test/config/AppConfigSpec.scala (hunk 1 above)
Comment (PatrykRudnicki) [reply test/config/AppConfigSpec.scala]: I agree with @ed0906 but moreove our countries are in json file so we should stay with json, we have support for csv and now when we add new stuff and we receive new stuff in csv we can use our new approach, but IMO we should support both things, read from Json and from csv, I'm not sure that we should convert all countries to csv, we're using this correctly.

File: app/services/HolderOfAuthenticationCode.scala (hunk 2 above)
Comment (PatrykRudnicki) [reply app/services/HolderOfAuthenticationCode.scala]: I think it's Holder of Authentication but inside contains Authorisation code. Just to clarify

File: test/services/HolderOfAuthorisationCodeSpec.scala [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+ * limitations under the License.
+ */
+
+package services
+
+import uk.gov.hmrc.play.test.UnitSpec
+
+class HolderOfAuthorisationCodeSpec extends UnitSpec {
+
+  "Holder of Authentication Code" should {
```
Comment (PatrykRudnicki) [inline test/services/HolderOfAuthorisationCodeSpec.scala]: So here we can change too, from `Holder of Authentication` to `Holder of Authorisation`

Comment (ed0906) [reply test/services/HolderOfAuthorisationCodeSpec.scala]: Fixed

**Pull request merged**